src/backend/utils/cache/inval.c

```diff
@@ -459,6 +460,7 @@ AddRelcacheInvalidationMessage(InvalidationMsgsGroup *group,
 	msg.rc.id = SHAREDINVALRELCACHE_ID;
 	msg.rc.dbId = dbId;
 	msg.rc.relId = relId;
+	msg.rc.local_only = get_ENR_withoid(currentQueryEnv, relId, ENR_TSQL_TEMP);
```

Comment:
can we skip adding entry and simply return? that way we do not need local_only flag. Also, let us have this check as a hook so that we can skip it for non-babelfish dbs

Reply:
The thing is that we *do* need local invalidation messages even for ENR tables. So it is necessary to allow the inval message to be created and saved, we just need to prevent it from being put in the shared inval queue.

Reply:
nit: we can skip calling `get_ENR_withoid()` if relid is invalid so no need to search for the entire array.

I noticed the callers of `RegisterRelcacheInvalidation()` have access to relation in a way to check if relation is a temp table. It might makes sense to consider changing the function signature in the future to pass an argument if table is temp so we could potentially skip the seq scan  if we're dealing with non-temp tables. But that is a bit invasive ofcourse.

My (nonblocking) concern is that we would also do the sequential search even for non-temporary tables if we do it this way. If a session has a lot of temp tables, then this would also degrade perf for all types of tables. This also wont be a no-op for non-babelfish instance either.